plutus-pab/src/Plutus/PAB/Arbitrary.hs

```diff
@@ -156,8 +157,53 @@ instance Arbitrary ThreadToken where
     shrink = genericShrink
 
 instance Arbitrary PlutusTx.Data where
-    arbitrary = genericArbitrary
-    shrink = genericShrink
+    arbitrary = sized arbitraryData
+      where
+        arbitraryData :: Int -> Gen PlutusTx.Data
+        arbitraryData n =
+          oneof [ arbitraryConstr n
+                , arbitraryMap n
+                , arbitraryList n
+                , arbitraryI
+                , arbitraryB
+                ]
+
+        arbitraryConstr 0 = do
+          (Positive ix) <- arbitrary
+          pure $ PlutusTx.Constr ix []
+        arbitraryConstr n = do
+          (n', m) <- segmentRange n
+          (Positive ix) <- arbitrary
+          args <- replicateM m (arbitraryData n')
+          pure $ PlutusTx.Constr ix args
+
+        arbitraryMap 0 = pure $ PlutusTx.Map []
+        arbitraryMap n = do
+          (n', m) <- segmentRange n
+          kvs <- replicateM m (arbitraryPair n')
+          pure $ PlutusTx.Map kvs
+
+        arbitraryPair n = do
+          (n', _) <- segmentRange n
+          (,) <$> arbitraryData n' <*> arbitraryData n'
```

Comment:
```suggestion
          (,) <$> arbitraryData n' <*> arbitraryData (n - n')
```
perhaps?

Reply:
I definitely recommend using `div` for size reductions.

Reply:
I ended up using div in this way, thanks @MaximilianAlgehed and @kozross